osu.Game.Tests/Visual/Multiplayer/TestSceneMultiSpectatorScreen.cs

```diff
@@ -80,6 +83,25 @@ public void TestGeneral()
             AddWaitStep("wait a bit", 20);
         }
 
+        [Test]
+        public void TestSpectatorPlayerInteractiveElementsHidden()
+        {
+            start(new[] { PLAYER_1_ID, PLAYER_2_ID });
+            loadSpectateScreen(false);
+
+            AddUntilStep("wait for player loaders", () => this.ChildrenOfType<PlayerLoader>().Count() == 2);
+            AddAssert("all player loader settings hidden", () => this.ChildrenOfType<PlayerLoader>().All(l => !l.ChildrenOfType<FillFlowContainer<PlayerSettingsGroup>>().Any()));
+
+            AddUntilStep("wait for players to load", () => spectatorScreen.AllPlayersLoaded);
+
+            // components wrapped in skinnable target containers load asynchronously, potentially taking more than one frame to load.
+            // therefore use until step rather than direct assert to account for that.
+            AddUntilStep("all interactive elements removed", () => this.ChildrenOfType<Player>().All(p =>
```

Comment:
This test step is consistently failing for me in test browser/non-headless.

Reply:
It passes locally for me? can you provide the thrown exception and the stack trace? weird it fails

Reply:
It just times out after the 50 repeats allowed. No stack trace.

![2021-08-14-190759_232x378_scrot](https://user-images.githubusercontent.com/20418176/129455008-675ce86f-8f35-4ec3-8c79-ec255d035e7b.png)

Reply:
Hmm, I'm still not sure what the cause of that could be, as I've tried multiple times with different skins and ended up passing the test. Would need you to debug this failure in that case, as I can't reproduce.

Reply:
If anything, I think it will be a client side setting (some kind of HUD toggle or otherwise).

Reply:
You're right, turns out hiding the HUD overlay blocks `SkinnableTargetContainer` from fininshing async load of components, as it schedules the lambda and never gets invoked until the HUD overlay gets updated.

Appears to be a trivial one to solve, will do and open another PR for that.

Reply:
In addition to https://github.com/ppy/osu/pull/14337, I've forced the HUD visibility mode to "Always" in the test case just to make sense of things.